### PR TITLE
Restore to use v1 api to fetch project level resources ancestry

### PR DESCRIPTION
**Pull request by iyabchen**

### CODE REVIEW

File: ancestrymanager/ancestrymanager.go [hunk 1]
```diff
@@ -35,15 +37,14 @@ type manager struct {
 // as value to the offline cache. If the key is not prefix with `projects/` or
 // `folders/`, it will be considered as a project. If no resourceManager is
 // provided, API requests for ancestry will be disabled.
-func New(resourceManager *cloudresourcemanager.Service, entries map[string]string, errorLogger *zap.Logger) (AncestryManager, error) {
-	return newManager(resourceManager, entries, errorLogger)
-}
-
-func newManager(resourceManager *cloudresourcemanager.Service, entries map[string]string, errorLogger *zap.Logger) (*manager, error) {
+func New(cfg *resources.Config, offline bool, entries map[string]string, errorLogger *zap.Logger) (AncestryManager, error) {
```
Comment (melinath) [inline ancestrymanager/ancestrymanager.go:40]: I am fine with us going back to taking cfg/offline as args now that we are going to have better tools for customizing the cfg's user agent and http client 👍 

Could you update the comment above to reflect this change?

File: ancestrymanager/ancestrymanager_test.go [hunk 2]
```diff
@@ -431,14 +456,21 @@ func TestGetAncestorsWithCache(t *testing.T) {
 				"folders/789":       {"folders/789", "organizations/321"},
 				"organizations/321": {"organizations/321"},
 			},
-			responses: map[string]*cloudresourcemanager.Project{
-				"projects/abc": {Name: "projects/123", ProjectId: "abc", Parent: "folders/456"},
-				"folders/456":  {Name: "folders/456", Parent: "folders/789"},
+			v3Responses: map[string]*crmv3.Project{
+				// "projects/abc": {Name: "projects/123", ProjectId: "abc", Parent: "folders/456"},
+				// "folders/456": {Name: "folders/456", Parent: "folders/789"},
```
Comment (melinath) [inline ancestrymanager/ancestrymanager_test.go:461]: double-checking - are these still here intentionally?

File: ancestrymanager/ancestrymanager_test.go [hunk 3]
```diff
@@ -348,77 +360,90 @@ func TestGetAncestors(t *testing.T) {
...
-				"projects/abc": {Name: "projects/123", ProjectId: "abc", Parent: "folders/456"},
-				"folders/456":  {Name: "folders/456", Parent: "folders/789"},
-				"folders/789":  {Name: "folders/789", Parent: "organizations/321"},
+			name:        "empty cache",
+			input:       "projects/abc",
+			cache:       make(map[string][]string),
+			v3Responses: map[string]*crmv3.Project{
+				// "projects/abc": {Name: "projects/123", ProjectId: "abc", Parent: "folders/456"},
+				// "folders/456": {Name: "folders/456", Parent: "folders/789"},
+				// "folders/789": {Name: "folders/789", Parent: "organizations/321"},
```
Comment (melinath) [inline ancestrymanager/ancestrymanager_test.go:434]: double-checking - are these still here intentionally?

File: ancestrymanager/ancestrymanager.go [hunk 4]
```diff
@@ -23,7 +24,8 @@ type manager struct {
 	errorLogger *zap.Logger
 	// GCP resource manager service. If this field is nil, online lookups will
```
Comment (melinath) [inline ancestrymanager/ancestrymanager.go:25]: could you update this comment?

**Review by melinath — APPROVED** [review]
LGTM and integration tests are passing 👍

A couple comments look like they may need to be updated to match the code changes.

File: ancestrymanager/ancestrymanager.go (hunk 4 above)
Comment (iyabchen) [reply ancestrymanager/ancestrymanager.go]: Done.

File: ancestrymanager/ancestrymanager_test.go (hunk 3 above)
Comment (iyabchen) [reply ancestrymanager/ancestrymanager_test.go]: Removed.

File: ancestrymanager/ancestrymanager_test.go (hunk 2 above)
Comment (iyabchen) [reply ancestrymanager/ancestrymanager_test.go]: Removed

File: ancestrymanager/ancestrymanager.go (hunk 1 above)
Comment (iyabchen) [reply ancestrymanager/ancestrymanager.go:42]: Done.

**Comment by iyabchen** [comment]
How to trigger testing on magic modules to test when folder permission is not available?

**Comment by melinath** [comment]
Do you mean you want to run the tests to see what happens if folder permission is not available? If so - you could make a PR that updates whitespace in a TFV file in magic modules after merging this one here.

**Pull request merged**